DOL.WHD.Section14c.Test/RepositoryMocks/ErrorLogRepositoryMock.cs

```diff
@@ -0,0 +1,76 @@
+
+        public bool Disposed => _disposed;
+
+
+        public ErrorLogRepositoryMock()
+        {
+            _data = new List<APIErrorLogs>
+            {
+                new APIErrorLogs {Id=1, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e02", EIN = "12-1234567", Exception = "My Test Exception 1", Level = "Fatal", Message="This a test message", User="test@test.com", UserId="12345"},
+                new APIErrorLogs {Id=2, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e03", EIN = "12-2234567", Exception = "My Test Exception 2", Level = "Error", Message="This a test message", User="test2@test.com", UserId="22345"},
+                new APIErrorLogs {Id=3, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e04", EIN = "12-3234567", Exception = "My Test Exception 3", Level = "Warn", Message="This a test message", User="test3@test.com", UserId="32345"},
+                new APIErrorLogs {Id=4, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e05", EIN = "12-4234567", Exception = "My Test Exception 4", Level = "Info", Message="This a test message", User="test4@test.com", UserId="42345"},
+                new APIErrorLogs {Id=5, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e06", EIN = "12-5234567", Exception = "My Test Exception 5", Level = "Debug", Message="This a test message", User="test5@test.com", UserId="52345"},
+                new APIErrorLogs {Id=6, CorrelationId="2edbc12f-4fd8-4fed-a848-b8bfff4d4e07", EIN = "12-6234567", Exception = "My Test Exception 6", Level = "Trace", Message="This a test message", User="test6@test.com", UserId="62345"},
+            };
+        }
+
+        public IQueryable<APIErrorLogs> GetAllLogs()
+        {
+            return _data.AsQueryable();
+        }
+
+        public LogDetails AddLog(LogDetails entity)
+        {
+            if (entity != null)
+            {
+                LogEventInfo eventInfo = new LogEventInfo();
+
+                eventInfo.Properties["EIN"] = string.IsNullOrEmpty(entity.EIN) ? string.Empty : entity.EIN;
+                eventInfo.LoggerName = "NLog";
+                if (string.IsNullOrEmpty(entity.Message))
+                {
+                    throw new ArgumentException("Message cannot be null or empty string", "Log Message");
+                }
+
+                eventInfo.Message = entity.Message;
+
+                if (!string.IsNullOrEmpty(entity.Exception))
+                {
+                    eventInfo.Exception = new Exception(entity.Exception);
+                    eventInfo.SetStackTrace((new System.Diagnostics.StackTrace(new Exception(entity.Exception), false)), 1);
+                }
+
+
+                eventInfo.Level = LogLevel.FromString(entity.Level);
+                eventInfo.Properties["UserId"] = entity.UserId;
+                eventInfo.Properties["UserName"] = entity.User;
+
+                eventInfo.Properties[Constants.IsServiceSideLog] = true;
+            }
+            return entity;
+        }
```

Comment:
Similar to the previous comment, it looks like having this just immediately `return entity` would satisfy the way the tests are written now.  However, there's a failure case in `ErrorLogsController.AddLog` that isn't exercised at all.  My first thought is possibly adding a toggle flag to the `ErrorLogRepositoryMock` class that sets whether `AddLog` returns or not.  E.g.:

```C#
public bool AddShouldFail { get; set; } = false;

public LogDetails AddLog(LogDetails entity)
{
    if (this.AddShouldFail)
    {
        return null;
    }
    return entity;
}
```

Then in `ErrorLogRepositoryTest.cs` (presumably to be renamed), you might have something like this:

```C#
[TestMethod]
[ExpectedException(typeof(ApiDataException),
    "Unable to add log")]
public void ErrorLog_AddLogs_Fails()
{
    // Arrange
    var service = new ErrorLogsController(_errorLogRepository);
    ((ErrorLogRepositoryMock)_errorLogRepository).AddShouldFail = true;

    // Act
    var result = service.AddLog(_data) as OkNegotiatedContentResult<LogDetails>; ;
}
```